client/src/main/scala/com/dwolla/cloudflare/AccessControlRuleClient.scala

```diff
@@ -0,0 +1,101 @@
+package com.dwolla.cloudflare
+
+import cats._
+import cats.effect._
+import cats.implicits._
+import com.dwolla.cloudflare.domain.dto.accesscontrolrules.AccessControlRuleDTO
+import com.dwolla.cloudflare.domain.dto.accounts.AccountDTO
+import com.dwolla.cloudflare.domain.model
+import com.dwolla.cloudflare.domain.model.{AccountId, Error, tagAccountId}
+import com.dwolla.cloudflare.domain.model.Exceptions.UnexpectedCloudflareErrorException
+import com.dwolla.cloudflare.domain.model.accesscontrolrules.Implicits._
+import com.dwolla.cloudflare.domain.model.accesscontrolrules._
+import io.circe.Json
+import io.circe.generic.auto._
+import io.circe.optics.JsonPath._
+import io.circe.syntax._
+import fs2._
+import org.http4s.Method._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.dsl.Http4sClientDsl
+
+import scala.language.higherKinds
+
+trait AccessControlRuleClient[F[_]] {
+  def list(accountId: AccountId): Stream[F, Rule]
+
+  def create(accountId: AccountId, rule: CreateRule): Stream[F, Rule]
+
+  def delete(accountId: AccountId, ruleId: String): Stream[F, String]
+}
+
+object AccessControlRuleClient {
+  def apply[F[_] : Sync](executor: StreamingCloudflareApiExecutor[F]): AccessControlRuleClient[F] = new AccessControlRuleClientImpl[F](executor)
+}
+
+class AccessControlRuleClientImpl[F[_] : Sync](executor: StreamingCloudflareApiExecutor[F]) extends AccessControlRuleClient[F] with Http4sClientDsl[F] {
+  def list(accountId: AccountId): Stream[F, Rule] =
+    for {
+      req ← Stream.eval(GET(BaseUrl / "accounts" / accountId / "firewall" / "access_rules" / "rules" withQueryParam("mode", "challenge")))
```

Comment:
Adding this query parameter is correct for our specific use case, but we'd like this library to be useful in general to access Cloudflare's API. I think we'd better serve that interest by making mode an optional parameter to this method.

What other modes are available?

If the Cloudflare API only supports a single `mode` parameter, then I'd start with an `Option[String]` parameter, but if the API supports multiple modes, we should probably start with `List[String]`.